include/deal.II/lac/scalapack.h

```diff
@@ -180,21 +180,106 @@ class ScaLAPACKMatrix : protected TransposeTable<NumberType>
-   * After this function is called, the content of the matrix is overwritten
-   * making it unusable.
+   * The eigenvectors are stored in the columns of the matrix, thereby
+   * overwriting the original content of the matrix.
+   *
+   * If all eigenvalues/eigenvectors have to be computed, pass the closed interval \f$ \left[ 0, M-1 \right] \f$ in @p index_limits.
+   *
+   * Pass the closed interval \f$ \left[ M-r, M-1 \right] \f$ if the \f$r\f$ largest eigenvalues/eigenvectors are desired.
    */
-  std::vector<NumberType> eigenvalues_symmetric();
+  std::vector<NumberType> eigenpairs_symmetric_by_index(const std::pair<unsigned int,unsigned int> &index_limits,
+                                                        const bool compute_eigenvectors);
+
+
 
   /**
-   * Compute all eigenpairs of a real symmetric matrix using <code>pXsyev</code>.
+   * Computing selected eigenvalues and, optionally, the eigenvectors.
+   * The eigenvalues/eigenvectors are selected by prescribing a range of values @p value_limits for the eigenvalues.
+   *
    * If successful, the computed eigenvalues are arranged in ascending order.
    * The eigenvectors are stored in the columns of the matrix, thereby
    * overwriting the original content of the matrix.
    */
-  std::vector<NumberType> eigenpairs_symmetric ();
+  std::vector<NumberType> eigenpairs_symmetric_by_value(const std::pair<NumberType,NumberType> &value_limits,
+                                                        const bool compute_eigenvectors);
+
+
+
+  /**
+  * Computing the singular value decomposition (SVD) of a
+  * matrix \f$A \in \mathbb{R}^{M \times N}\f$, optionally computing the left and/or right
+  * singular vectors. The SVD is written as \f$A = U * \Sigma * V^T\f$
+  * with \f$\Sigma \in \mathbb{R}^{M \times N}\f$ as a diagonal matrix,
+  * \f$U \in \mathbb{R}^{M \times M}\f$ and \f$U \in \mathbb{R}^{M \times M}\f$
+  * as orthogonal matrices. The diagonal elements of \f$\Sigma\f$
+  * are the singular values of \f$A\f$ and the columns of \f$U\f$ and \f$V\f$ are the
+  * corresponding left and right singular vectors, respectively. The
+  * singular values are returned in decreasing order and only the first \f$\min(M,N)\f$
+  * columns of \f$U\f$ and rows of VT = \f$V^T\f$ are computed.
+  *
+  * Upon return the content of the matrix is unusable.
+  * The matrix A must have identical block cyclic distribution for the rows and column.
+  *
+  * If left singular vectors are required matrices \f$A\f$ and \f$U\f$
+  * have to be constructed with the same process grid and block cyclic distribution.
+  * If right singular vectors are required matrices \f$A\f$ and \f$V^T\f$
+  * have to be constructed with the same process grid  and block cyclic distribution.
+  *
+  * To avoid computing the left and/or right singular vectors the function accepts <code>nullptr</code>
+  * for @p U and/or @p VT.
+  */
+  std::vector<NumberType> compute_SVD(ScaLAPACKMatrix<NumberType> *U,
```

Comment:
maybe make them default, i.e. `ScaLAPACKMatrix<NumberType> *U = nullptr` and `ScaLAPACKMatrix<NumberType> *VT = nullptr`?